code/modules/mob/living/carbon/xenomorph/update_icons.dm

```diff
@@ -52,14 +52,23 @@
 		return
 
 	if(stat == DEAD)
-		icon_state = "[mutation_type] [caste.caste_type] Dead"
+		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Dead"
+		if(!icon_state) in icon_states(icon_xeno)
```

Comment:
what is this trying to do

```
	if(stat == DEAD)
		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Dead"
		if(!(icon_state in icon_states(icon_xeno)))
			icon_state = "Normal [caste.caste_type] Dead"
	else if(lying)
		if((resting || sleeping) && (!knocked_down && !knocked_out && health > 0))
			icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Sleeping"
			if(!(icon_state in icon_states(icon_xeno)))
				icon_state = "Normal [caste.caste_type] Sleeping"
		else
			icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Knocked Down"
			if(!(icon_state in icon_states(icon_xeno)))
				icon_state = "Normal [caste.caste_type] Knocked Down"
	else
		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Running"
		if(!(icon_state in icon_states(icon_xeno)))
			icon_state = "Normal [caste.caste_type] Running"
```


this?

Reply:
Check to see if the icon even exists. If not use the caste's default sprite

Reply:
okay but it doesn't compile with `(!icon_state) in icon_states(icon_xeno)`, `!(icon_state in icon_states(icon_xeno)`

Reply:
yeah, it should work now